executor/src/witgen/machines/block_machine.rs

```diff
@@ -138,6 +142,9 @@ impl<'a, T: FieldElement> BlockMachine<'a, T> {
             data,
             publics: Default::default(),
             first_in_progress_row: block_size,
+            multiplicity_counter: MultiplicityCounter::new(
```

Comment:
Here we clone part of `parts` so it's stored twice in `BlockMachine`. I'm wondering if there is a cleaner solution. I don't have a clear suggestion yet however.

Reply:
Maybe adding an `increment_multiplicity` method to the `Machine` trait? Then we can access the `parts` of `self` here and don't need to store them inside `multiplicity_counter`

Reply:
I'm not sure how adding `increment_multiplicity` to the `Machine` trait would help? Who stores the (stateful) multiplicity counter?

Adding this to the machine trait would also be a bit strange because some machines (e.g. Memory) can only be connected via a permutation, so they never need multiplicity counting.

Reply:
Right that doesn't work. Seems like using `Connection` helps a lot already. Closing